### PR TITLE
add esoteric platforms to validate adhoc/release pipelines

### PR DESCRIPTION
**Pull request by evanahlberg**
## Description

Add esoteric platforms to adhoc/release pipelines.
Revert changes to verify pipeline to use containers again now that chef-foundation is installed on esoteric platforms.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: omnibus/config/projects/chef.rb [hunk 1]
```diff
@@ -45,27 +45,27 @@
...
+# # addons which require omnibus software defns (not direct deps of chef itself - RFC-063)
+# #
+# dependency "nokogiri" # (nokogiri cannot go in the Gemfile, see wall of text in the software defn)
 
-# FIXME?: might make sense to move dependencies below into the omnibus-software chef
-#  definition or into a chef-complete definition added to omnibus-software.
-dependency "gem-permissions"
+# # FIXME?: might make sense to move dependencies below into the omnibus-software chef
+# #  definition or into a chef-complete definition added to omnibus-software.
+# dependency "gem-permissions"
```
Comment (gcs-devel) [inline omnibus/config/projects/chef.rb:60]: Delete comments

File: omnibus/config/projects/chef.rb [hunk 2]
```diff
@@ -45,27 +45,27 @@
...
+# dependency "chef"
 
-#
-# addons which require omnibus software defns (not direct deps of chef itself - RFC-063)
-#
-dependency "nokogiri" # (nokogiri cannot go in the Gemfile, see wall of text in the software defn)
+# #
+# # addons which require omnibus software defns (not direct deps of chef itself - RFC-063)
+# #
+# dependency "nokogiri" # (nokogiri cannot go in the Gemfile, see wall of text in the software defn)
```
Comment (gcs-devel) [inline omnibus/config/projects/chef.rb:57]: Delete comments

File: omnibus/config/projects/chef.rb [hunk 3]
```diff
@@ -45,27 +45,27 @@
 overrides_path = File.expand_path("../../../../omnibus_overrides.rb", current_file)
 instance_eval(IO.read(overrides_path), overrides_path)
```
Comment (gcs-devel) [inline omnibus/config/projects/chef.rb:46]: Let's remove this line and remove `the `omnibus_overrides.rb` file. The override files have been moved to https://github.com/chef/chef-foundation for chef-16 and chef-18

File: omnibus/Gemfile.lock [hunk 4]
```diff
@@ -8,8 +8,8 @@ GIT
```
Comment (gcs-devel) [inline omnibus/Gemfile.lock:8]: We need a `bundle update --conservative` here to get the new version of license_scout

File: omnibus/Gemfile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 source "https://rubygems.org"
 
-gem "omnibus", github: ENV.fetch("OMNIBUS_GITHUB_REPO", "chef/omnibus"), branch: ENV.fetch("OMNIBUS_GITHUB_BRANCH", "main")
+gem "omnibus", github: ENV.fetch("OMNIBUS_GITHUB_REPO", "chef/omnibus"), branch: "debug-mac-signing"
```
Comment (gcs-devel) [inline omnibus/Gemfile:3]: We can revert this change now

File: omnibus/Gemfile.lock (hunk 4 above)
Comment (gcs-devel) [reply omnibus/Gemfile.lock:8]: Please revert the gemfile before doing the `bundle update` so we get omnibus back to the correct branch as well.

File: .buildkite/build-test-omnibus.sh [hunk 6]
```diff
@@ -21,92 +26,262 @@ for platform in ${platforms[@]}; do
...
-done
+fi
+
+if [ $BUILDKITE_PIPELINE_SLUG == "chef-chef-main-validate-release" ]
+then
+  echo "- wait: ~"
+  echo "- key: create-build-record"
+  echo "  label: \":artifactory: Create Build Record\""
+  echo "  plugins:"
+  echo "  - chef/omnibus#v0.2.83:"
```
Comment (jesseprieur) [inline .buildkite/build-test-omnibus.sh:186]: We need to align these versions throughout. We have some using 0.2.81 and others using 0.2.83

File: .buildkite/hooks/pre-command [hunk 7]
```diff
@@ -3,23 +3,23 @@
 set -eu
 
 # Only execute in the verify pipeline
-[[ "$BUILDKITE_PIPELINE_NAME" =~ verify$ ]] || exit 0
+[[ "$BUILDKITE_PIPELINE_NAME" =~ verify$ ]] || [[ "$BUILDKITE_PIPELINE_NAME" =~ validate/release$ ]] || [[ "$BUILDKITE_PIPELINE_NAME" =~ validate/adhoc$ ]]  || [[ "$BUILDKITE_PIPELINE_NAME" =~ validate/adhoc-canary$ ]] || exit 0
```
Comment (jesseprieur) [inline .buildkite/hooks/pre-command:6]: Should we do this as a proper regex? `(verify|validate/(release|adhoc|canary))$`

File: .buildkite/hooks/pre-command [hunk 8]
```diff
@@ -3,23 +3,23 @@
...
 export OMNIBUS_TOOLCHAIN_VERSION
-echo $OMNIBUS_TOOLCHAIN_VERSION
+echo "Omnibus Toolchain Version: $OMNIBUS_TOOLCHAIN_VERSION"
 
-if [ $BUILDKITE_STEP_KEY == "build-windows-2019" ] && [ $BUILDKITE_ORGANIZATION_SLUG == "chef" ]
+if [ $BUILDKITE_STEP_KEY == "build-windows-2019" ] && [[ "$BUILDKITE_ORGANIZATION_SLUG" =~ chef(-canary)?$ ]]
 then
   TOKEN=$(curl -sX PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
-  RESPONSE=$(curl -sH "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/iam/security-credentials/default-windows-2019-privileged-chef-Role)
+  RESPONSE=$(curl -sH "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/iam/security-credentials/core-default-windows-2019-privileged-$BUILDKITE_ORGANIZATION_SLUG-Role)
```
Comment (jesseprieur) [inline .buildkite/hooks/pre-command:22]: @evanahlberg was there a way to do this more dynamically?

File: .buildkite/hooks/pre-command (hunk 7 above)
Comment (evanahlberg) [reply .buildkite/hooks/pre-command:6]: on main i did this.. forgot to change it: 

`[[ "$BUILDKITE_PIPELINE_NAME" =~ verify$ ]] || [[ "$BUILDKITE_PIPELINE_NAME" =~ validate/.* ]] || exit 0`

**Review by tpowell-progress — APPROVED** [review]

File: .buildkite/build-test-omnibus.sh [hunk 9]
```diff
@@ -21,92 +28,262 @@ for platform in ${platforms[@]}; do
 done
 
 # remove duplicates from build array
-omnibus_build_platforms=($(printf "%s\n" "${omnibus_build_platforms[@]}" | sort -u | tr '\n' ' '))
+if [[ ! -z "${omnibus_build_platforms:-}" ]]
```
Comment (tpowell-progress) [inline .buildkite/build-test-omnibus.sh:31]: I think `-z` should handle an unset variable comparison, but also quotes should force to a zero length string, so the `:-` feels redundant?
